motto/thread_agents/thread_agents.py

```diff
@@ -82,92 +85,111 @@ def is_empty_message(self, message):
+            message = self.messages.get()
+            if message.message == self.stop_message:
+                break
+            self.said = False
+            self.set_canceling_variable(False)
+            self.set_interrupt_variable(False)
+            for r in self.message_processor(message):
+                self.outputs.put(r)
+
     def message_processor(self, input: AgentArgs):
         '''
         Takes a single message and provides a response if no canceling event has occurred.
         '''
-        if  self.is_empty_message(input.message):
+        if self.is_empty_message(input.message):
             message = None
-        elif str(input.message).startswith("(") and  str(input.message).endswith(")"):
+        elif str(input.message).startswith("(") and str(input.message).endswith(")"):
             message = input.message
-        else :
+        else:
             message = f"(Messages (user \"{input.message}\"))"
-        if( message is None) and (input.additional_info is None):
+        if (message is None) and (input.additional_info is None):
             self.set_processing_val(False)
             return
-        response = super().__call__(message, input.functions, input.additional_info).content
+
         self.set_processing_val(True)
-        self.handle_event()
-        resp = None
+        response = super().__call__(message, input.functions, input.additional_info).content
         for resp in self.process_stream_response(response):
-            self.handle_event()
-            #cancel processing of the current message and return the message to the input
+            # cancel processing of the current message and return the message to the input
             if self.cancel_processing_var:
-                self.log.info(f"message_processor:cancel processing for message {message}")
+                self.log.info(f"message_processor:cancel processing for message {message}\n")
                 self.input(input.message)
                 break
 
             if self.interrupt_processing_var:
-                self.log.info(f"message_processor:interrupt processing for message {message}")
+                self.log.info(f"message_processor:interrupt processing for message {message}\n")
                 resp = "..."
 
             self.history += [E(S(assistant_role), G(ValueObject(resp)))]
             self.log.info(f"message_processor: return response for message {message} : {resp}")
             yield resp if input.language is None else (resp, input.language)
-            #interrupt processing
+            # interrupt processing
             if self.interrupt_processing_var:
                 break
         self.set_processing_val(False)
 
-
     def __call__(self, msgs_atom=None, functions=[], additional_info=None):
         if msgs_atom is not None:
-            super().input(AgentArgs(msgs_atom, functions, additional_info))
+            self.messages.put(AgentArgs(msgs_atom, functions, additional_info))
```

Comment:
So, `__call__` puts its input to `messages` directly, and `messages_processor` runs in its own thread. Another solution would possibly to create a subscription in the `event_bus` and put messages there and get these messages from the bus by `messages_processor` (which would be a callback for this subscription). In this case, there would be no need to run `messages_processor` in a separate thread. `__call__` would just put messages there instead of `messages` variable. It it'd work, it would be a better solution